src/main/kotlin/hwr/oop/projects/peakpoker/core/hand/HandEvaluator.kt

```diff
@@ -1,47 +1,83 @@
 package hwr.oop.projects.peakpoker.core.hand
 
 import hwr.oop.projects.peakpoker.core.card.Card
+import hwr.oop.projects.peakpoker.core.card.HoleCards
+import hwr.oop.projects.peakpoker.core.card.CommunityCards
 import hwr.oop.projects.peakpoker.core.card.Rank
+import hwr.oop.projects.peakpoker.core.hand.HandRank
 
-/**
- * Evaluates a poker hand and returns its ranking.
- */
 object HandEvaluator {
+
     /**
-     * Determines the ranking of the given list of five cards.
-     * @param list of five cards representing the hand
-     * @return the corresponding HandRank
+     * Evaluates the best HandRank from exactly five cards.
      */
-    fun evaluate(cards: List<Card>): HandRank {
-        // Hand must contain exactly five cards and all must be unique
+    internal fun evaluate(cards: List<Card>): HandRank {
         require(cards.size == 5) { "Hand must contain exactly 5 cards" }
-        require(cards.toSet().size == 5) { "Hand must contain 5 unique cards" }
+        require(cards.distinct().size == 5) { "Hand must contain 5 unique cards" }
 
```

Comment:
If you still want to use cards as the val to store the community cards in the evaluation process, you could use them like this:
```suggestion
val cards = communityCards.cards
```